### ISSUE TITLE
[CI][Packaging][Java] java-jars was broken by updating bundled aws-sdk-cpp

### ISSUE THREAD

**Issue by kou** [apache/arrow#34111]
### Describe the bug, including details regarding any error messages, version, and platform.

#33808 broke java-jars job:

https://github.com/apache/arrow/pull/33808#issuecomment-1424873183

https://github.com/ursacomputing/crossbow/actions/runs/4138641959/jobs/7155286836#step:6:4265

```text
Undefined symbols for architecture x86_64:
  "_SSLClose", referenced from:
      _s_handle_shutdown in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLCopyPeerTrust", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLCreateContext", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLHandshake", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLRead", referenced from:
      _s_process_read_message in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLSetCertificate", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLSetConnection", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLSetIOFuncs", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLSetPeerDomainName", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLSetProtocolVersionMin", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLSetSessionOption", referenced from:
      _s_tls_handler_new in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SSLWrite", referenced from:
      _s_process_write_message in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SecCertificateCreateWithData", referenced from:
      _aws_import_public_and_private_keys_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
      _aws_import_trusted_certificates in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecIdentityCreateWithCertificate", referenced from:
      _aws_import_public_and_private_keys_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecItemExport", referenced from:
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
  "_SecItemImport", referenced from:
      _aws_import_ecc_key_into_keychain in libaws-c-io.a(darwin_pki_utils.c.o)
      _aws_import_public_and_private_keys_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecKeyCopyPublicKey", referenced from:
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_SecKeyCreateRandomKey", referenced from:
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
  "_SecKeyCreateSignature", referenced from:
      _s_sign_message in libaws-c-cal.a(securityframework_ecc.c.o)
  "_SecKeyCreateWithData", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_SecKeyVerifySignature", referenced from:
      _s_verify_signature in libaws-c-cal.a(securityframework_ecc.c.o)
  "_SecKeychainCopyDefault", referenced from:
      _aws_import_public_and_private_keys_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecKeychainOpen", referenced from:
      _aws_import_public_and_private_keys_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecKeychainUnlock", referenced from:
      _aws_import_public_and_private_keys_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecPKCS12Import", referenced from:
      _aws_import_pkcs12_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_SecPolicyCreateBasicX509", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SecPolicyCreateSSL", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SecTrustEvaluate", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SecTrustSetAnchorCertificates", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SecTrustSetAnchorCertificatesOnly", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_SecTrustSetPolicies", referenced from:
      _s_drive_negotiation in libaws-c-io.a(secure_transport_tls_channel_handler.c.o)
  "_kSecAttrCanDerive", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrCanSign", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrCanVerify", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrKeyClass", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrKeyClassPrivate", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrKeyClassPublic", referenced from:
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrKeySizeInBits", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrKeyType", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecAttrKeyTypeECSECPrimeRandom", referenced from:
      _aws_ecc_key_pair_new_from_private_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_public_key_impl in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_generate_random in libaws-c-cal.a(securityframework_ecc.c.o)
      _aws_ecc_key_pair_new_from_asn1 in libaws-c-cal.a(securityframework_ecc.c.o)
  "_kSecImportExportPassphrase", referenced from:
      _aws_import_pkcs12_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_kSecImportItemIdentity", referenced from:
      _aws_import_pkcs12_to_identity in libaws-c-io.a(darwin_pki_utils.c.o)
  "_kSecKeyAlgorithmECDSASignatureDigestX962", referenced from:
      _s_sign_message in libaws-c-cal.a(securityframework_ecc.c.o)
      _s_verify_signature in libaws-c-cal.a(securityframework_ecc.c.o)
```

https://github.com/ursacomputing/crossbow/actions/runs/4138641959/jobs/7155287119#step:6:1325

```text
FAILED: s2n_tls_ep-prefix/src/s2n_tls_ep-stamp/s2n_tls_ep-configure D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-stamp/s2n_tls_ep-configure 
cmd.exe /C "cd /D D:\a\crossbow\crossbow\arrow\cpp-build\cpp\s2n_tls_ep-prefix\src\s2n_tls_ep-build && "C:\Program Files\CMake\bin\cmake.exe" -P D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-stamp/s2n_tls_ep-configure-RELEASE.cmake && "C:\Program Files\CMake\bin\cmake.exe" -E touch D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-stamp/s2n_tls_ep-configure"
CMake Error at D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-stamp/s2n_tls_ep-configure-RELEASE.cmake:37 (message):
  Command failed: 1

   'C:/Program Files/CMake/bin/cmake.exe' '-DCMAKE_C_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe' '-DCMAKE_CXX_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe' '-DCMAKE_AR=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/lib.exe' '-DCMAKE_RANLIB=:' '-DBUILD_SHARED_LIBS=OFF' '-DBUILD_STATIC_LIBS=ON' '-DBUILD_TESTING=OFF' '-DCMAKE_BUILD_TYPE=RELEASE' '-DCMAKE_CXX_FLAGS=/DWIN32 /D_WINDOWS  /GR /EHsc /D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING' '-DCMAKE_CXX_FLAGS_DEBUG=/MDd /Zi /Ob0 /Od /RTC1' '-DCMAKE_CXX_FLAGS_MISIZEREL=/MD /O1 /Ob1 /DNDEBUG' '-DCMAKE_CXX_FLAGS_RELEASE=/MD /O2 /Ob2 /DNDEBUG' '-DCMAKE_CXX_FLAGS_RELWITHDEBINFO=/MD /Zi /O2 /Ob1 /DNDEBUG' '-DCMAKE_CXX_STANDARD=17' '-DCMAKE_C_FLAGS=/DWIN32 /D_WINDOWS /W3' '-DCMAKE_C_FLAGS_DEBUG=/MDd /Zi /Ob0 /Od /RTC1' '-DCMAKE_C_FLAGS_MISIZEREL=/MD /O1 /Ob1 /DNDEBUG' '-DCMAKE_C_FLAGS_RELEASE=/MD /O2 /Ob2 /DNDEBUG' '-DCMAKE_C_FLAGS_RELWITHDEBINFO=/MD /Zi /O2 /Ob1 /DNDEBUG' '-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=' '-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=' '-DCMAKE_INSTALL_LIBDIR=lib' '-DCMAKE_VERBOSE_MAKEFILE=FALSE' '-DCMAKE_C_COMPILER_LAUNCHER=D:/a/crossbow/crossbow/sccache/sccache.exe' '-DCMAKE_CXX_COMPILER_LAUNCHER=D:/a/crossbow/crossbow/sccache/sccache.exe' '-DCMAKE_BUILD_TYPE=RELEASE' '-DENABLE_TESTING=OFF' '-DENABLE_UNITY_BUILD=ON' '-DCMAKE_INSTALL_PREFIX=D:/a/crossbow/crossbow/arrow/cpp-build/cpp/awssdk_ep-install' '-DCMAKE_PREFIX_PATH=D:/a/crossbow/crossbow/arrow/cpp-build/cpp/awssdk_ep-install' '-GNinja' 'D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep'

  See also

    D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-stamp/s2n_tls_ep-configure-*.log


-- stdout output is:
-- The C compiler identification is MSVC 19.29.30147.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detected CMAKE_SYSTEM_PROCESSOR as AMD64
-- Detected 64-Bit system
-- The ASM compiler identification is MSVC
-- Found assembler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
-- Enabling Kyber_R3 x86_64 optimizations
-- Disabling stacktrace functionality
-- Configuring incomplete, errors occurred!
See also "D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-build/CMakeFiles/CMakeOutput.log".
See also "D:/a/crossbow/crossbow/arrow/cpp-build/cpp/s2n_tls_ep-prefix/src/s2n_tls_ep-build/CMakeFiles/CMakeError.log".

-- stderr output is:
CMake Error at C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find crypto (missing: crypto_LIBRARY)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/Findcrypto.cmake:75 (find_package_handle_standard_args)
  CMakeLists.txt:398 (find_package)
```

### Component(s)

Continuous Integration, Java, Packaging

**Comment by js8544** [comment]
OpenSSL seems to be causing a lot of issues. We can also configure awssdk to use AWS::crypto from [aws-lc](https://github.com/awslabs/aws-lc) instead of OpenSSL's libcrypto. I can experiment with it later today. cc @kou

**Comment by kou** [comment]
Thanks!
Can we use aws-lc and OpenSSL in the same process? (No symbol confliction?)

**Comment by js8544** [comment]
> Thanks! Can we use aws-lc and OpenSSL in the same process? (No symbol confliction?)

s2n provides an options to rename all aws-lc symbols to avoid conflicts: https://github.com/aws/s2n-tls/blob/main/CMakeLists.txt#L495. It works on my machine. I've submitted a pr to see if it works: https://github.com/apache/arrow/pull/34159

**Comment by kou** [comment]
Fixed by #34159.
